lib/ex_doc/formatter/html.ex

```diff
@@ -343,15 +343,27 @@ defmodule ExDoc.Formatter.HTML do
       skip_undefined_reference_warnings_on: config.skip_undefined_reference_warnings_on
     ]
 
-    config.extras
-    |> Task.async_stream(
-      &build_extra(&1, groups, language, autolink_opts, source_url_pattern),
-      timeout: :infinity
-    )
-    |> Enum.map(&elem(&1, 1))
+    extras =
+      config.extras
+      |> Task.async_stream(
+        &build_extra(&1, groups, language, autolink_opts, source_url_pattern),
+        timeout: :infinity
+      )
+      |> Enum.map(&elem(&1, 1))
+
+    ids_count = Enum.reduce(extras, %{}, &Map.update(&2, &1.id, 1, fn c -> c + 1 end))
+
+    extras
+    |> Enum.with_index(fn extra, idx ->
+      if ids_count[extra.id] > 1, do: disambiguate_id(extra, idx + 1), else: extra
```

Comment:
This version of `with_index` is not supported in earlier Elixir versions. Maybe implement this with a `Enum.map_reduce` instead?

Reply:
ExDoc requires Elixir 1.11, I think we're good. :)

Reply:
not sure if I got it, shall I use other function than `with_index`, there are some other places in the codebase that seems to use it just fine, like

https://github.com/elixir-lang/ex_doc/blob/b66861bb60173b056022e95662cf72275eba18a8/lib/ex_doc/formatter/epub.ex#LL91C18-L91C18

But indeed the tests for 1.11 are failing

Reply:
This particular version of `with_index` does not work, so you need to use something else. I would do seomthing like this:
```
Enum.map_reduce(extras, 1, fn extra, idx ->
  if ids_count[extra.id] > 1, do: {disam(...), idx + 1}, else: {extra, idx}
end)
|> elem(0)
```